tensorflow/c/c_api.h

```diff
@@ -1137,14 +1146,18 @@ TF_CAPI_EXPORT extern void TF_AbortWhile(const TF_WhileParams* params);
 // shapes in `y`.
 // The partial derivatives are returned in `dy`. `dy` should be allocated to
 // size `nx`.
+// `prefix` names the scope into which all gradients operations are being added.
+// If `prefix` is nullptr, "gradients" is used by default.
```

Comment:
We may not use `prefix/` exactly, right? We might end up using `<prefix>_1/` etc.
Though, I guess that still means that operations are prefixed with `prefix` :)

I fear that this behavior may be confusing (e.g., one calls `TF_AddGradientsWithPrefix` with prefix `foo` and expects to analyze nodes added to the graph by looking for `foo/` as a prefix of the operation names, but doesn't find anything because the prefix is actually `foo_1`.

I think if it would be clearer/better for `TF_AddGradientsWithPrefix` to:

1. Fail on a NULL `prefix`.
2. Fail if `prefix` clashes with an existing prefix.

While `TF_AddGradients` retains its behavior (since it doesn't make any promises on the names of nodes added).

Without this behavior, I feel things may get hard to reason about (for example, in Java, one could create two `Scope` objects on the same `Graph` and easily mess up names). This will also be consistent with the behavior of other C API methods that add a node to the graph (`TF_FinishOperation` will fail if the name isn't unique).